pine64/pinebook-pro/sound/reset-sound.rb

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env nix-shell
```

Comment:
Shouldn't be this installed? Does the sound reset need to be applied on every boot?

Reply:
> Does the sound reset need to be applied on every boot?

No, I remember to have used that script only one time (maybe the first time I installed NixOS on the Pinebook Pro). Maybe there are some conditions that make necessary to run it again. Or maybe it's simply not necessary anymore. Since I was undecided I put it here exactly as it was in `wip-pinebook-pro`.

> Shouldn't be this installed?

Since I'm not even sure if it has to be executed I think it can be left as is. Otherwise i could add it to the overlay to make it easier to install.

@samueldr Do you know when is necessary to run this script?

Reply:
It is necessary to run only once to set things up correctly. Running it more than once will *reset* audio parameters.

This something that, AFAIUI, should be handled by ALSA UCM normally, but I don't know if there is any effort in the community to handle that.

Other distros do something that I personally think is kinda worse: they have a copy of the `/var/lib/alsa/asound.state` file that they pre-seed.